### ISSUE TITLE
Unknown type of `requests.Session().headers`

### ISSUE THREAD

**Issue by vrslev** [microsoft/pyright#2799]
**Describe the bug**
Pyright complains about type of `requests.Session()` headers attribute:
```console
$ pyright t.py       
No configuration file found.
pyproject.toml file found at /private/tmp.
Loading pyproject.toml file at /private/tmp/pyproject.toml
Assuming Python platform Darwin
Auto-excluding **/node_modules
Auto-excluding **/__pycache__
Auto-excluding .git
stubPath /private/tmp/typings is not a valid directory.
Searching for source files
Found 1 source file
/private/tmp/t.py
  /private/tmp/t.py:4:1 - error: Type of "headers" is partially unknown
    Type of "headers" is "CaseInsensitiveDict[Unknown]" (reportUnknownMemberType)
1 error, 0 warnings, 0 infos 
Completed in 0.627sec
```
Mypy doesn't show any errors, so I guess that problem has nothing to do with typeshed:
```console
$ mypy t.py           
Success: no issues found in 1 source file
```

**To Reproduce**
1. Create Python virtual environment, activate it and install `requests`,
2. Create Python file with following content:
```python
import requests

session = requests.Session()
session.headers
```
3. Create Pyright config file with `typeCheckingMode` set to `strict`,
4. Run Pyright against Python file.

**VS Code extension or command-line**
Pylance doesn't show any errors, release v2021.12.2.
CLI does. I tested with multiple versions and type check fails with version 1.1.201.

**Comment by erictraut** [comment]
Am I correct in assuming that you're running with "strict" mode enabled? I don't see any error here using the default configuration.

In strict mode, the libraries you import must have complete type information. The `requests` type stubs have some "typing holes" in them that lead to partial type information. In this particular case, the typeshed stub includes the following type alias definition:

```python
CaseInsensitiveDict = structures.CaseInsensitiveDict
```

Because the type arguments have been omitted in the RHS, this is assumed to be:
```python
CaseInsensitiveDict = structures.CaseInsensitiveDict[Any, Any]`
```

But pyright treats all omitted type arguments as `Unknown`, a special form of `Any`.

If you want this to be fixed, please consider submitting a bug report — or better yet, a PR — to typeshed.

**Comment by vrslev** [comment]
> Am I correct in assuming that you're running with "strict" mode enabled?

Yes, I am. 

> The requests type stubs have some "typing holes" in them that lead to partial type information.

I totally get that. Why then nothing was failing before version 1.1.201?

**Comment by erictraut** [comment]
Versions prior to 1.1.201 contained a long-standing bug that I fixed. It is now working as intended.

**Comment by vrslev** [comment]
Okay! A will send a PR to typeshed then. Thanks! @erictraut

**Comment by erictraut** [comment]
Thanks! Let me know when the PR is merged, and I'll make it a priority to pull the updated stubs into pyright.

**Comment by vrslev** [comment]
I looked at stubs for requests and there's nothing wrong with them.

Also I created repository — https://github.com/vrslev/pyright-alias-issue — that shows when type check passes and when fails: https://github.com/vrslev/pyright-alias-issue/compare/passes. Don't pay attention to naming 😃. 

Note that this is only relevant to version >1.1.201. Both examples pass on version 1.1.199.

**Comment by vrslev** [comment]
```python
CaseInsensitiveDict = structures.CaseInsensitiveDict
```
later used in
```python
    headers: CaseInsensitiveDict[Text]
```
where `structures.CaseInsensitiveDict` is
```python
class CaseInsensitiveDict(MutableMapping[str, _VT], Generic[_VT]):
    ...
```

**Comment by erictraut** [comment]
The problem is that `CaseInsensitiveDict = structures.CaseInsensitiveDict` is defining a type alias, but `structures.CaseInsensitiveDict` is a generic type. Because there is no type argument provided, `structures.CaseInsensitiveDict` is correctly interpreted to mean `structures.CaseInsensitiveDict[Any]`. 

I think the intent here was to define a generic type alias, in which case it needs to be defined as such:
```python
_VT = TypeVar("_VT")
CaseInsensitiveDict = structures.CaseInsensitiveDict[_VT]
```

Or, since the name is the same, the stub could simply re-export the symbol without redefining it.

```python
from .structures import CaseInsensitiveDict as CaseInsensitiveDict
```

**Comment by vrslev** [comment]
@erictraut Latest release fixed the issue. Thanks!
